agent/eventhandler/task_handler.go

```diff
@@ -15,244 +15,323 @@ package eventhandler
 
 import (
 	"container/list"
+	"context"
 	"errors"
+	"fmt"
 	"sync"
 	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
 	"github.com/aws/amazon-ecs-agent/agent/statechange"
 	"github.com/aws/amazon-ecs-agent/agent/statemanager"
 	"github.com/aws/amazon-ecs-agent/agent/utils"
 	"github.com/cihub/seelog"
 )
 
-// Maximum number of tasks that may be handled at once by the TaskHandler
-const concurrentEventCalls = 3
+const (
+	// concurrentEventCalls is the maximum number of tasks that may be handled at
+	// once by the TaskHandler
+	concurrentEventCalls = 3
 
-type eventList struct {
-	// events is a list of *sendableEvents
-	events *list.List
-	// sending will check whether the list is already being handlerd
-	sending bool
-	//eventsListLock locks both the list and sending bool
-	eventListLock sync.Mutex
-}
+	// drainEventsFrequency is the frequency at the which unsent events batched
+	// by the task handler are sent to the backend
+	drainEventsFrequency = time.Minute
```

Comment:
Should we go faster than a minute?  A minute seems pretty slow to me.

Reply:
As per offline conversation, i'll change this to `20s`